### PR TITLE
Add Jupyter repr and default interpreter repr methods

### LINKED ISSUE #127: Make Jupyter-aware with _repr_html_()
### Is your feature request related to a problem? Please describe.

When using PrettyTable in Jupyter Notebooks, print(table) produces the ASCII table which is inferior to an HTML representation like pandas and other notebook-friendly libraries use.

A pedantic solution to using the HTML output would be:

```python
from IPython.display import HTML, display

display(HTML(table.get_html_string()))
```

But giving PrettyTable a proper `_repr_html_()` works much better: `PrettyTable._repr_html_ = PrettyTable.`

image

### Describe the solution you'd like

PrettyTable should have a `_repr_html_()` which calls `self.get_html_string()` so that it is a Jupyter-aware table.
PrettyTable should also have a basic `__repr__()` method returning the same as `__str__()` while we're at it. Not having a `repr` at all seems pretty lacking compared to most mature libraries.

### PR DESCRIPTION
**Pull request by amirali**
Fixes #127.

I added the features they talked about them in #127.

### CODE REVIEW

**Review by hugovk — CHANGES_REQUESTED** [review]
Thank you! Please could you add tests to cover these lines?

**Comment by amirali** [comment]
I added the test for new repr and Jupyter environment repr. Sorry I didn't add them at first

File: tests/test_prettytable.py [hunk 1]
```diff
@@ -1592,3 +1592,19 @@ def my_format(col: str, value: Any) -> str:
...
+
+@pytest.fixture
+def simple_table():
+    pt = PrettyTable(["field1", "field2"])
+    pt.add_row(["row1_field1", "row1_field2"])
+    pt.add_row(["row2_field1", "row2_field2"])
+    return pt
+
+
+class TestRepr:
```
Comment (kumaraditya303) [inline tests/test_prettytable.py:1597]: A test for `__getattr__` with `_repr_html_` is required, Thank you

**Review by kumaraditya303 — DISMISSED** [review]

File: tests/test_prettytable.py (hunk 1 above)
Comment (amirali) [reply tests/test_prettytable.py:1597]: Written here 921bdbe

**Comment by flutefreak7** [comment]
This looks good to me! I'm the author of #127

File: tests/test_prettytable.py [hunk 2]
```diff
@@ -1592,3 +1592,25 @@ def my_format(col: str, value: Any) -> str:
 +-------------+----------+-----------+------------+
 """.strip()
         )
+
+
+@pytest.fixture
+def simple_table():
```
Comment (hugovk) [inline tests/test_prettytable.py]: Can we reuse an existing fixture instead of creating a new one?

**Comment by hugovk** [comment]
I'm not sure if my earlier review comments saved properly, will try again!

---

Do we need this as an attribute?

```python
        elif name == "_repr_html_":
            return self._repr_html_()
```

Looks like IPython only needs the method, not attribute:

* https://ipython.readthedocs.io/en/stable/config/integrating.html#rich-display

**Comment by amirali** [comment]
When I implemented the method, in the first runs in Jupyter it doesn't show the expected value in HTML and got AttriubuteError on calling the method. So I thought it should be registered as an attribute. But if you think it's not necessary I'll double-check it and remove the extra part.

**Comment by hugovk** [comment]
I'm not super familiar with IPython / JupyterLab but seemed to work fine for me without the attribute:

image

Let's remove it, and if it's needed later it's easy to add. Thanks!

**Comment by amirali** [comment]
Done.

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
Thank you!

**Pull request merged**